components/ws-manager-api/core.proto

```diff
@@ -446,6 +446,12 @@ message WorkspaceMetadata {
     // Annotations are key/value pairs that gets attached to the workspace.
     // This is primarily intended for annotating headless workspace loads.
     map<string, string> annotations = 4;
+
+    // team the workspace belongs to
+    string team = 5;
+
+    // project the workspace belongs to
+    string project = 6;
```

Comment:
That's a good question! Here is how I understand it: In contrast to proto2, proto3 uses no presence semantics as default (i.e. it just does not serialize values that are not present) and does not use explicit presence semantics (i.e. it stores if a field has explicitly been set to a value). 

Tagging the fields with optional would switch it to explicit presence semantics.  Explicit presence therefore increases message size, even when no value is present, because of the tracking requirement. In return additional methods are generated that allow you to check if a value has been set. I do not foresee a need to do this, as with no presence semantics we get an empty string for team/project which should be good enough for us. What do you think?

Reply:
> I do not foresee a need to do this, as with no presence semantics we get an empty string for team/project which should be good enough for us. What do you think?

Sometimes `optional` is nice for readability bc a) you get the `has_` accessor and b) for making it explicit in the proto file. (Or if you need the extra bit in case you need the default value in the value domain - not the case here),
But searching our repo we mention optional [mostly](https://github.com/gitpod-io/gitpod/blob/e3423ccb96223de8f0ac83ca9f51317a8f3e20b9/components/public-api/gitpod/v1/workspaces.proto#L268) on comments, so fine with not using it! :+1:  A comment would be nice, though. :slightly_smiling_face: